MREUnityRuntime/MREUnityRuntimeLib/Core/ColliderGeometry.cs

```diff
@@ -119,21 +122,98 @@ private void Patch(BoxCollider collider)
     public class MeshColliderGeometry : ColliderGeometry
     {
         /// <inheritdoc />
-        public override ColliderType ColliderType => ColliderType.Mesh;
+        public override ColliderType Shape => ColliderType.Mesh;
+
+        public Guid MeshId { get; set; }
+        public bool Convex { get; set; }
```

Comment:
>public bool Convex { get; set; } [](start = 8, length = 32)

Curreently all mesh colliders are convex as that is the only way that a rigidbody can be added and will work on an actor.  Should this even be here?  We had decided not to support non-convex meshes until we had introduced a concept of static objects that couldn't and shouldn't allow adding of rigidbodies.

Reply:
+1

---
In reply to: [316739876](https://github.com/microsoft/mixed-reality-extension-unity/pull/98#discussion_r316739876) [](ancestors = 316739876)

Reply:
This is more for futureproofing. Currently all instances of this class have Convex = true, but if it's a feature we could eventually support, it doesn't cost us anything to have it in there.

---
In reply to: [316767033](https://github.com/microsoft/mixed-reality-extension-unity/pull/98#discussion_r316767033) [](ancestors = 316767033,316739876)

Reply:
TBH I forgot this was in here, but I still like being explicit. My first question when I see a mesh collider in code is, convex or not?

---
In reply to: [316826712](https://github.com/microsoft/mixed-reality-extension-unity/pull/98#discussion_r316826712) [](ancestors = 316826712,316767033,316739876)

Reply:
I still think that this is something that should be added when it is needed.  When/if we add static collision objects that can have non-convex meshes, it should be added then, possibly not even here if there is a different way we implement the feature.  Adding this in to our code even know means that a developer in our runtime will think that whether a mesh is convex or not is something that is a choice within our runtime.

---
In reply to: [316848494](https://github.com/microsoft/mixed-reality-extension-unity/pull/98#discussion_r316848494) [](ancestors = 316848494,316826712,316767033,316739876)